cmd/watch.go

```diff
@@ -0,0 +1,85 @@
+package cmd
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+
+	"github.com/redhat-developer/odo/pkg/application"
+	"github.com/redhat-developer/odo/pkg/component"
+	"github.com/redhat-developer/odo/pkg/project"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var watchLocal string
+
+var watchCmd = &cobra.Command{
+	Use:   "watch [component name]",
+	Short: "Watch for changes, update component on change",
+	Long:  `Watch for changes, update component on change.`,
+	Example: `  # Watch for changes in directory for current component
+  odo watch
+
+  # Watch for changes in directory for component called frontend 
+  odo watch frontend
+	`,
+	Args: cobra.MaximumNArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		stdout := os.Stdout
+		client := getOcClient()
+		projectName := project.GetCurrent(client)
+		applicationName, err := application.GetCurrent(client)
+		checkError(err, "Unable to get current application.")
+
+		var componentName string
+		if len(args) == 0 {
+			var err error
+			log.Debug("No component name passed, assuming current component")
+			componentName, err = component.GetCurrent(client, applicationName, projectName)
+			if err != nil {
```

Comment:
we don't need to do `if err != nil` while using `checkError()`

Reply:
thx, fixed